spark_pipeline_framework_testing/test_runner.py

```diff
@@ -346,15 +346,22 @@ def write_table_to_output(
         df: DataFrame = spark_session.table(view_name)
         types: List[Tuple[str, Any]] = df.dtypes
         type_dict: Dict[str, Any] = {key: value for key, value in types}
-        if "array" in type_dict.values() or "struct" in type_dict.values():
+        # these type strings can look like 'array<struct<Field:string>>', so we
+        # have to check if "array" or "struct" appears in the type string, not
+        # just for exact matches
+        if [t for t in type_dict.values() if "array" in t or "struct" in t]:
```

Comment:
This line probably deserves extra scrutiny. It works, but it feels like it could be written more clearly.

Reply:
Probably good enough for now since it is looking in column types which is a limited set